app/parse.py

```diff
@@ -15,9 +32,70 @@ class Product:
     num_of_reviews: int
 
 
+def init_driver(headless: bool = True) -> webdriver.Chrome:
+    options = Options()
+    options.headless = headless
+    return webdriver.Chrome(options=options)
+
+
+def parse_single_product(prod_soup: Tag) -> Product:
+    return Product(
+        title=prod_soup.select_one(".title")["title"],
+        description=prod_soup.select_one(".description").text,
+        price=float(prod_soup.select_one(".price").text.replace("$", "")),
+        rating=int(prod_soup.select_one("p[data-rating]")["data-rating"]),
+        num_of_reviews=int(prod_soup.select_one(".review-count").text.split()[0])
+    )
+
+
+def get_products(driver: webdriver.Chrome, url: str) -> list:
+    driver.get(url)
+    accept_cookies(driver)
+
+    while True:
+        try:
+            button = driver.find_element(By.CLASS_NAME, "ecomerce-items-scroll-more")
+            WebDriverWait(driver, 10).until(ec.element_to_be_clickable(button)).click()
+            WebDriverWait(driver,10).until(ec.presence_of_all_elements_located((By.CLASS_NAME, "card-body")))
+        except NoSuchElementException:
+            break
+        except TimeoutException:
+            break
```

Comment:
While the exception handling for `NoSuchElementException` and `TimeoutException` is correct, it might be beneficial to log these exceptions to understand why they occurred. This can help in identifying issues with the page loading or element identification.